### PR TITLE
Add support for missing modalities

### PR DESCRIPTION
**Pull request by Sllambias**
Same PR as previous. Fixed based on comments.
The 1 major thing that is still happening here that you did not like is the object array which I dont see how we avoid in this case.

Otherwise just improved and now tested and passing tests without the cursed task conversion file

### CODE REVIEW

File: yucca/training/data_loading/YuccaDataset.py [hunk 1]
```diff
@@ -116,6 +123,44 @@ def _transform(self, data_dict, case):
             data_dict = self.composed_transforms(data_dict)
         return self.to_torch(data_dict)
 
+    def replace_missing_modalities_with_zero_arrays(self, data):
```
Comment (asbjrnmunk) [inline yucca/training/data_loading/YuccaDataset.py:126]: this function is no longer being called

File: yucca/training/data_loading/YuccaDataset.py [hunk 2]
```diff
@@ -116,6 +123,44 @@ def _transform(self, data_dict, case):
...
+        # in classification this avoids setting the zero array to the 1d array with classes
+        sizes = [i.size for i in data]
+        idx_largest_array = np.where(sizes == np.max(sizes))[0][0]
+        for idx, i in enumerate(data):
+            if i.size == 0:
+                data[idx] = np.zeros(data[idx_largest_array].squeeze().shape)
+        return data
+
+    def unpack(self, data):
+        if self.supervised:
```
Comment (asbjrnmunk) [inline yucca/training/data_loading/YuccaDataset.py:138]: As a rule of thumb, i think the readability is impacted negatively when a method accesses data which is not provided directly as an argument. I would propose providing `self.supervised` with a type annotation as an argument and only use the properties when it is absolutely necessary.

File: yucca/training/data_loading/YuccaDataset.py [hunk 3]
```diff
@@ -116,6 +123,44 @@ def _transform(self, data_dict, case):
...
+        sizes = [i.size for i in data]
+        idx_largest_array = np.where(sizes == np.max(sizes))[0][0]
+
+        # replace missing modalities with zero-filed arrays
+        for idx, i in enumerate(data):
+            if i.size == 0:
+                data[idx] = np.zeros(data[idx_largest_array].squeeze().shape)
+
+        # unpack array into images and (maybe) labels
+        if self.supervised:
```
Comment (asbjrnmunk) [inline yucca/training/data_loading/YuccaDataset.py:156]: Same as above

**Review by asbjrnmunk — APPROVED** [review]
Looks good! I have a minor comment below

File: yucca/training/data_loading/YuccaDataset.py (hunk 1 above)
Comment (Sllambias) [reply yucca/training/data_loading/YuccaDataset.py:126]: good catch

**Pull request merged**